docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -0,0 +1,475 @@
+ml_train_model_set_dataset(model, dataset);
+```
+
+## Construct a model
+
+A model can be constructed with `ml_train_model_construct()`.
+If you have a file that describes the model, the file can be used to construct initially with `ml_train_model_construct_with_file()`.
+Even if the model is constructed from a file, switching, modifying, or setting a component is possible only until you compile with `ml_train_model_compile()`.
+
+### Construct a model from a description file
+
+As of now, only INI formatted files `*.ini` is supported to construct a model from a file.
+
+#### Create a model from INI formatted file
+
+Special sections `[Model]` and `[Dataset]` are respectively referring to `model` and `dataset` object.
+Rest of INI sections map to a `layer`. Keys and values from each section set properties of the layer.
+All keys and values are treated as case-insensitive.
+
+For example:
+
+```ini
+[Model] # Special section that describes model itself
+Type = NeuralNetwork  # Model Type : only NeuralNetwork is supported as of now
+Optimizer = adam  # Optimizer : sgd (stochastic gradient decent),
+          #             adam (Adaptive Moment Estimation)
+#### optimizer related properties
+Learning_rate = 0.0001  # Learning Rate for the optimizer
+Decay_rate = 0.96 # The decay rate for decaying the learning rate
+Decay_steps = 1000       # decay step for the exponentially decayed learning rate
+beta1 = 0.9     # beta 1 for adam
+beta2 = 0.9999  # beta 2 for adam
+epsilon = 1e-7  # epsilon for adam
+
+#### Model compile related properties
+batch_size = 9
+loss = cross      # Cost(loss) function : mse (mean squared error)
+                        #                       cross (cross entropy)
+
+####  Model run related properties
+Epochs = 20     # Epochs
+save_path = "model.bin"   # model path to save and read parameters
+
+[DataSet] # Special section that describes dataset.
+BufferSize=9
+TrainData="trainingSet.dat"
+ValidData="validationSet.dat"
+LabelData="label.dat"
+
+# Layer Sections, each section name refers to name of the layer
+[inputlayer]
+Type = input
+Input_Shape = 1:1:62720 # Input Layer Dimension in channel:height:width
+Normalization = true
+
+[outputlayer]
+Type = fully_connected
+Unit = 2    # Output Layer Dimension ( Weight Width )
+bias_initializer = zeros
+weight_initializer = xavier_uniform
+Activation = sigmoid  # activation : sigmoid, softmax
+weight_regularizer = l2norm
+weight_regularizer_constant = 0.005
+```
+
+The following restrictions must be adhered to:
+
+ - Model file must have a `[Model]` section.
+ - Model file must have at least one layer.
+ - Valid keys must have valid properties. The invalid keys in each section is ignored.
+
+> [!NOTE]
+> All paths are relative to the current working directory unless stating an absolute path.
+> It is recommended to set `save_path` and `Dataset` from the code over describing inside the model file.
```

Comment:
@zhoonit, yes the connation will be the same. Suggest changing as suggested by @mijin85cho:

Set `save_path` and `Dataset` from the code rather than describing inside the model file.

Reply:
Done